src/expr.rs

```diff
@@ -854,19 +829,58 @@ fn rewrite_call(context: &RewriteContext,
                 -> Option<String> {
     debug!("rewrite_call, width: {}, offset: {}", width, offset);
 
-    // FIXME using byte lens instead of char lens (and probably all over the place too)
     // 2 is for parens
-    let max_callee_width = try_opt!(width.checked_sub(2));
-    let callee_str = try_opt!(callee.rewrite(context, max_callee_width, offset));
-    debug!("rewrite_call, callee_str: `{}`", callee_str);
-
-    if args.is_empty() {
-        return Some(format!("{}()", callee_str));
+    let mut hi = try_opt!(width.checked_sub(2)) * 2;
+    let mut lo = 0;
+    let mut tries = 0;
+
+    // Binary search for the best split between callee and arguments.
```

Comment:
Could you add to the comment: what exactly are you trying to optimise? In what cases will this succeed/fail on the first attempt? Do you ever expect it to take >10 tries or is that just for safety? (btw 10 seems pretty pessimistic, given that there is only a 100 chars max, I'd expect to get a decent result in something like 2 or 3 tries or never (although, I don't quite understand completely what is going on here). Also, is it possible to go with a 'reasonable' answer after n tries, rather than giving up completely if we don't get a perfect answer?